lisa/sut_orchestrator/qemu/platform.py

```diff
@@ -131,11 +136,19 @@ def _configure_node_capabilities(
             return True
 
         host_capabilities = self._get_host_capabilities(qemu_conn, log)
+        nodes_capabilities = self._create_node_capabilities(host_capabilities)
 
         nodes_requirement = []
         for node_space in environment.runbook.nodes_requirement:
-            node_capabilities = self._create_node_capabilities(log, node_space)
-            nodes_requirement.append(node_capabilities)
+            # Check that the general node capabilities are compatible with this node's
+            # specific requirements.
+            if not node_space.check(nodes_capabilities):
```

Comment:
I think I finally figured out how node capabilities/requirements are supposed to be handled. Does this look correct?

Reply:
Yes, it's right. And the capability is extendable, if there are special things in qemu to enforce like vm_size in Azure.